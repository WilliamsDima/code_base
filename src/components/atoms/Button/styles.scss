@import '../../../assets/styles/colors.scss';
@import '../../../assets/styles/globalStyles.scss';


.button {
    padding: 10px;
    background-color: $mainColor;
    border-radius: 5px;
    text-transform: uppercase;
    font-weight: 700;
    font-size: 15px;
    color: $white;
}