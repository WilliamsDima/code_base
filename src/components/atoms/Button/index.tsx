import React, { FC } from "react"
import './styles.scss';

interface IButton {
  onClick?: () => void
  children: React.ReactNode
}

const Button: FC<IButton> = ({onClick, children}) => {


  return (
    <button className="button" onClick={onClick}>
        {children}
    </button>
  )
}

export default Button
