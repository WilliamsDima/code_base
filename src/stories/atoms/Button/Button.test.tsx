import Button from './index'
import { render, screen } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'

describe('Компанента Button', () => {
  test('Renter Button with children', () => {
    render(<Button>Button</Button>)
    const textChildren = screen.getByText(/Button/i)
    const notIsText = screen.queryByText(/some text/i)
    const btn = screen.getByRole('button')
    expect(textChildren).toBeInTheDocument()
    expect(notIsText).toBeNull()
    expect(btn).toBeInTheDocument()
    expect(btn).toMatchSnapshot()
  })
})
