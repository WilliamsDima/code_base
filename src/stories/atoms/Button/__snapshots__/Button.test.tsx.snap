// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Компанента Button Renter Button with children 1`] = `
<button
  class="btn shadowClick"
>
  Button
</button>
`;
